rawhttp-core/src/main/java/rawhttp/core/RequestLine.java

```diff
@@ -88,16 +105,38 @@ private void writeTo(OutputStream outputStream, boolean newLine) throws IOExcept
         outputStream.write(method.getBytes(StandardCharsets.US_ASCII));
         outputStream.write(' ');
 
-        String path = uri.getRawPath();
-        if (path == null || path.isEmpty()) {
-            outputStream.write('/');
+        //RFC-7230 section 5.3.3
+        if ("CONNECT".equalsIgnoreCase(method)) {
```

Comment:
I think the easier way to do this logic would be to already check the options here, like this:
```java
if (!options.allowIllegalConnectAuthority() && "CONNECT.equalsIgnoreCase(method)) ...
```
There's no need to enter this branch in any other case. Also notice the code below is checking the wrong option.

Reply:
Jeez, good catch, starts with allowIllegal and they all looked the same 🙈